rpki-validator/src/main/java/net/ripe/rpki/validator3/api/bgp/BgpPreviewService.java

```diff
@@ -238,16 +229,82 @@ public static BgpPreviewEntry of(Asn origin, IpRange prefix, Validity validity)
             return sorting.getDirection() == Sorting.Direction.DESC ? columns.reversed() : columns;
         }
 
+        public abstract IpRange getPrefix();
+        abstract BgpPreviewEntry ofValidity(Validity validity);
+
+        public Asn getOrigin() {
+            return new Asn(Integer.toUnsignedLong(origin));
+        }
+
+    }
+
+    @lombok.Value
+    @lombok.EqualsAndHashCode(callSuper = true)
+    @lombok.ToString(callSuper = true)
+    public static class BgpPreviewEntry4 extends BgpPreviewEntry {
+        short prefixLength;
+        int prefix;
+
+        private BgpPreviewEntry4(int origin, Validity validity, short prefixLength, int prefix) {
+            super(origin, validity);
+            this.prefixLength = prefixLength;
+            this.prefix = prefix;
+        }
+
+        public static BgpPreviewEntry4 of(Asn origin, Validity validity, IpRange prefix) {
+            return new BgpPreviewEntry4((int) origin.longValue(), validity, (short) prefix.getPrefixLength(), (int) ((Ipv4Address) prefix.getStart()).longValue());
+        }
+
+        public IpRange getPrefix() {
+            return IpRange.prefix(new Ipv4Address(Integer.toUnsignedLong(prefix)), prefixLength);
+        }
+
         BgpPreviewEntry ofValidity(Validity validity) {
-            return new BgpPreviewEntry(origin, prefix, validity);
+            return new BgpPreviewEntry4(origin, validity, prefixLength, prefix);
         }
 
-        public Asn getOrigin() {
-            return new Asn(origin);
+    }
+
+    @lombok.Value
+    @lombok.EqualsAndHashCode(callSuper = true)
+    @lombok.ToString(callSuper = true)
+    public static class BgpPreviewEntry6 extends BgpPreviewEntry {
+        private static BigInteger TWO_64 = BigInteger.ONE.shiftLeft(64);
+        private static BigInteger TWO_64_MINUS_1 = TWO_64.subtract(BigInteger.ONE);
+
+        short prefixLength;
+        long prefixHi;
```

Comment:
You're splitting the 128 bits of the prefix into two longs. It would be clearer for me if there was a small piece of documentation here that indicated that is what this class is doing :)

Reply:
Added documentation.